src/hexkit/providers/mongodb/__init__.py

```diff
@@ -18,6 +18,6 @@
 and associated utilities.
 """
 
-from .provider import MongoDbConfig, MongoDbDaoFactory  # noqa: F401
+from .provider import MongoDbConfig, MongoDbDaoFactory
```

Comment:
Absolute import here too (unless there's a reason not to do that in an __init__.py)